compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -113,7 +113,7 @@ class Erasure extends Phase with DenotTransformer {
       }
     case ref: JointRefDenotation =>
       new UniqueRefDenotation(
-        ref.symbol, transformInfo(ref.symbol, ref.symbol.info), ref.validFor, ref.prefix)
+        ref.symbol, transformInfo(ref.symbol, ref.symbol.info), ref.validFor, ref.prefix, ref.isRefinedMethod)
```

Comment:
This isRefinedMethod flag only seems needed for JointRefDenotations, nothing else can be a refinement, and after erasure there's no refinement so it shouldn't be needed either.

Reply:
I checked this and for the following
```scala
package a

object A {

  trait Fn[-T1, +R] { def apply(v1: T1): R }

  val fn0: Fn[Int, Int] { def apply(arg: Int): Int } = ???

  val fn = fn0

}
```
referencing `fn.apply` in another compilation unit has a `UniqueRefDenotation`

Reply:
ok so yeah it seems a refinement that only modifies argument names is `UniqueRefDenotation` but something that changes the return type will be `JointRefDenotation`

Reply:
the fix is probably to force in `goRefined` a new `JointRefDenotation` as it was likely leading to a new allocation each time anyway - as it is the only place to set `isRefinedMethod = true`. Or else check in `meet` for method parameter name changes

Reply:
i have made a change where anytime `isRefinedMethod` is true for a copy then make a `JointRefDenotation`